Clicker/SectionControllers/PollDateSectionController.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  PollDateSectionController.swift
+//  Clicker
+//
+//  Created by Kevin Chan on 8/31/18.
+//  Copyright © 2018 CornellAppDev. All rights reserved.
+//
+
+import IGListKit
+
+protocol PollDateSectionControllerDelegate {
+    var role: UserRole { get }
+}
+
+enum PollDateSectionControllerItemType {
+    case date
+    case card
+}
+
+class PollDateSectionController: ListSectionController {
+    
+    // MARK: - Data Vars
+    var pollDateModel: PollDateModel!
+    var delegate: PollDateSectionControllerDelegate!
+    var itemTypes: [PollDateSectionControllerItemType]!
+    
+    // MARK: - Constants
+    let dateCellHeight: CGFloat = 50
+    
+    init(delegate: PollDateSectionControllerDelegate) {
+        self.delegate = delegate
+    }
+    
+    // MARK: - ListSectionController overrides
+    override func numberOfItems() -> Int {
+        return itemTypes.count
+    }
+    
+    override func sizeForItem(at index: Int) -> CGSize {
+        guard let containerSize = collectionContext?.insetContainerSize else {
+            return .zero
+        }
+        let itemType = itemTypes[index]
+        switch (itemType) {
+        case .date:
+            return CGSize(width: containerSize.width, height: dateCellHeight)
+        case .card:
+            let cardCellHeight = calculateCardCellHeight(poll: pollDateModel.poll, userRole: delegate.role)
+            return CGSize(width: containerSize.width, height: cardCellHeight)
+        }
+    }
+    
+    override func cellForItem(at index: Int) -> UICollectionViewCell {
+        var cell: UICollectionViewCell
+        let itemType = itemTypes[index]
+        switch (itemType) {
+        case .date:
+            let dateCell = collectionContext?.dequeueReusableCell(of: DateCell.self, for: self, at: index) as! DateCell
+            dateCell.configure(with: pollDateModel.date)
+            dateCell.setNeedsUpdateConstraints()
+            cell = dateCell
+            break
```

Comment:
do you need break here? I forget if its actually needed

Reply:
oh you don't. force of habit from last semester haha. ill remove it